rotkehlchen/db/drivers/server.py

```diff
@@ -29,74 +27,63 @@ class DBConnectionType(Enum):
-    rowcount: int
-    lastrowid: int
-
-
-class SerializedZMQReturnData(TypedDict):
-    """Dictionary type for serialized ZMQ return data"""
-    result: SerializedDBWriteResult | None
-    error: Any
-
-
-class DbMethod(SerializableEnumNameMixin):
-    INITIALIZE = auto()
-    OPEN_CURSOR = auto()
-    CLOSE_CURSOR = auto()
-    EXECUTE = auto()
-    EXECUTEMANY = auto()
-    EXECUTESCRIPT = auto()
-    ROLLBACK = auto()
-    COMMIT = auto()
-    CLOSE = auto()
-
-
-@dataclass
-class ZMQCallData:
-    """The data packet sent through ZMQ call"""
-    connection_type: DBConnectionType
-    db_path: str
-    cursor_name: str | None
-    method: DbMethod
-    args: Sequence[Any]
-    kwargs: dict[str, Any]
-
-
-class DBWriteResult(TypedDict):
-    rowcount: int
-    lastrowid: int
-
-
-@dataclass()
-class ZMQReturnData:
-    """The data packet received from ZMQ call"""
-    result: DBWriteResult | None
-    error: BaseException | None
-
-    def serialize(self) -> SerializedZMQReturnData:
-        return {
-            'result': self.result,
-            'error': self.error,
-        }
-
-    @classmethod
-    def deserialize(cls, data: SerializedZMQReturnData) -> 'ZMQReturnData':
-        return cls(
-            result=data['result'],
-            error=data['error'],
-        )
+DBMethod = db_messages_pb2.DBMethod  # type: ignore[attr-defined]  # pylint: disable=no-member
+DBWriteResult = db_messages_pb2.DBWriteResult()  # type: ignore[attr-defined]  # pylint: disable=no-member
+DBWriterError = db_messages_pb2.DBWriterError()  # type: ignore[attr-defined]  # pylint: disable=no-member
+ZMQCallData = db_messages_pb2.ZMQCallData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+ZMQReturnData = db_messages_pb2.ZMQReturnData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+
+
+def to_typed_data(data: Any) -> db_messages_pb2.TypedData:  # type: ignore[name-defined]  # pylint: disable=no-member
+    """Converts a python object to a protobuf TypedData object
+    Supported types: list, tuple, bytes, str, bool, int, float, and None"""
+    typed_data = db_messages_pb2.TypedData()  # type: ignore[attr-defined]  # pylint: disable=no-member
+    if isinstance(data, list | tuple):
+        typed_data.type = 'array'
+        typed_array = db_messages_pb2.TypedArray()  # type: ignore[attr-defined]  # pylint: disable=no-member
+        typed_array.array.extend([to_typed_data(item) for item in data])
+        typed_data.array.CopyFrom(typed_array)
+    elif isinstance(data, bytes):
+        typed_data.type = 'bytes'
+        typed_data.bytes = data
+    elif isinstance(data, str):
+        typed_data.type = 'string'
+        typed_data.bytes = data.encode()
+    elif isinstance(data, bool):
+        typed_data.type = 'bool'
+        typed_data.bytes = data.to_bytes()
+    elif isinstance(data, int):
+        typed_data.type = 'int'
+        typed_data.bytes = data.to_bytes(length=32, byteorder='big', signed=True)
```

Comment:
why are you setting the lenght as 32? Why not 64 or any other value? What drove this decision?